src/trix/models/composition.coffee

```diff
@@ -78,16 +78,22 @@ class Trix.Composition extends Trix.BasicObject
     block = document.getBlockAtIndex(index)
 
     if block.getBlockBreakPosition() is offset
-      document = document.removeTextAtRange(range)
-      range = [position, position]
+      if block.text.getStringAtRange([offset - 1, offset]) is "\n"
+        document = document.removeTextAtRange([position - 1, position])
+      else if offset - 1 isnt 0
+        position += 1
     else
       if block.text.getStringAtRange([offset, offset + 1]) is "\n"
         range = [position - 1, position + 1]
       else if offset - 1 isnt 0
         position += 1
 
     newDocument = new Trix.Document [block.removeLastAttribute().copyWithoutText()]
-    @setDocument(document.insertDocumentAtRange(newDocument, range))
+    if @getBlock()?.getConfig("breakOnReturn")
+      document = document.removeTextAtRange([position - 1, position])
+      @setDocument(document.insertDocumentAtRange(newDocument, [position - 1, position]))
```

Comment:
Is it possible to avoid reassigning `document` and just reassign `range` instead, removing the `else` clause and letting it fall through to a single `@setDocument` call? Would that be clearer?